packages/dataviews/src/normalize-fields.ts

```diff
@@ -43,10 +43,13 @@ export function normalizeFields< Item >(
 
 		const renderFromElements = ( { item }: { item: Item } ) => {
 			const value = getValue( { item } );
-			return (
-				field?.elements?.find( ( element ) => element.value === value )
-					?.label || getValue( { item } )
-			);
+			const label = field?.elements?.find( ( element ) => {
+				// Intentionally using == here to allow for type coercion.
+				// eslint-disable-next-line eqeqeq
+				return element.value == value;
```

Comment:
Isn't the issue in the field though. If the field is "string" its values should be string both in `getValue` and `elements` for me. and same for integer no?

Reply:
https://github.com/WordPress/gutenberg/pull/64391 may be a better approach.